src/ebonite/core/objects/wrapper.py

```diff
@@ -28,52 +35,126 @@ class ModelWrapper(EboniteParams):
     Must be pyjackson-serializable
     """
     type = None
+    methods_json = 'methods.json'
 
     def __init__(self):
         self.model = None
+        self.methods: typing.Optional[Methods] = None
 
-    @abstractmethod
+    @contextlib.contextmanager
     def dump(self) -> FilesContextManager:
+        with self._dump() as artifact:
+            yield artifact + Blobs({self.methods_json: InMemoryBlob(dumps(self.methods).encode('utf-8'))})
+
+    @abstractmethod
+    def _dump(self) -> FilesContextManager:
         """
         Must return context manager with :class:`~ebonite.core.objects.ArtifactCollection`
         :return: :class:`~ebonite.core.objects.ArtifactCollection`
         """
         pass
 
-    @abstractmethod
     def load(self, path):
+        self._load(path)
+        with open(os.path.join(path, self.methods_json), 'r', encoding='utf-8') as f:
+            self.methods = loads(f.read(), typing.Optional[Methods])
+
+    @abstractmethod
+    def _load(self, path):
         """
         Must load and set internal .model field
         :param path:
         :return:
         """
         pass
 
-    def bind_model(self, model):
+    def bind_model(self, model, **kwargs):
         """
-        Bind model object to this wrapper
+        Bind model object to this wrapper by using given input data sample
 
         :param model: model object to bind
+        :param kwargs: additional information to be used for analysis
         :return: self
         """
+        input_data = kwargs.get('input_data', None)
+        if input_data is None:
```

Comment:
i think you can add input_data to function arguments explicitly in this case

Reply:
fixed